base/sparse/higherorderfns.jl

```diff
@@ -871,11 +874,11 @@ promote_containertype(::Type{AbstractSparseArray}, ::Type{Tuple}) = Array
 
 # broadcast[!] entry points for combinations of sparse arrays and other (scalar) types
 @inline function broadcast_c{N}(f, ::Type{AbstractSparseArray}, mixedargs::Vararg{Any,N})
-    parevalf, passedargstup = capturescalars(f, mixedargs)
+    parevalf, passedargstup = capturescalars(f, map(_sparsifystructured, mixedargs))
```

Comment:
> I have thought of having a method that iterates over columns for AbstractArrays in general and doing something like _broadcast! with _broadcast_getindex! so we don't need this mechanism

Could you clarify? If you mean something along the lines of extending the common interface over `SparseVector`s and `SparseMatrixCSC`s in sparse `broadcast[!]` to other types (e.g. scalars, thereby obviating the `capturescalars` mechanism), then that matches my plan for evolving these as well.